### DISCUSSION TITLE
App and script naming in "generate", capitalization and underscores, getting sidetracked

### DISCUSSION THREAD

**Discussion by westmj** [mojolicious/mojo#1680] [Q&A] [answered]
Very new to Mojolicious and web frameworks.... maybe getting off track, but trying to work this through... 

Can someone please explain the logic behind how I get messages/files containing things like:

west_Mj_01  from  "mojo generate app west_MJ_01" (notice J and j)
my_app from "mojo generate app MyApp"
and
w_e_s_t__m_j_01.yml from "mojo generate app WEST_MJ_01"

Is the simplification to only use non-capitalized letters? No underscores? 

(bionic)westm@localhost:~/google_drive/mojo$ mojo generate app west_MJ_01
  [mkdir] /home/westm/google_drive/mojo/west_Mj_01/script
  [write] /home/westm/google_drive/mojo/west_Mj_01/script/west_Mj_01
  [chmod] /home/westm/google_drive/mojo/west_Mj_01/script/west_Mj_01 744
  [mkdir] /home/westm/google_drive/mojo/west_Mj_01/lib
  [write] /home/westm/google_drive/mojo/west_Mj_01/lib/west_MJ_01.pm
  [exist] /home/westm/google_drive/mojo/west_Mj_01
  [write] /home/westm/google_drive/mojo/west_Mj_01/west_MJ_01.yml
  [mkdir] /home/westm/google_drive/mojo/west_Mj_01/lib/west_MJ_01/Controller
  [write] /home/westm/google_drive/mojo/west_Mj_01/lib/west_MJ_01/Controller/Example.pm
  [mkdir] /home/westm/google_drive/mojo/west_Mj_01/t
  [write] /home/westm/google_drive/mojo/west_Mj_01/t/basic.t
  [mkdir] /home/westm/google_drive/mojo/west_Mj_01/public
  [write] /home/westm/google_drive/mojo/west_Mj_01/public/index.html
  [mkdir] /home/westm/google_drive/mojo/west_Mj_01/templates/layouts
  [write] /home/westm/google_drive/mojo/west_Mj_01/templates/layouts/default.html.ep
  [mkdir] /home/westm/google_drive/mojo/west_Mj_01/templates/example
  [write] /home/westm/google_drive/mojo/west_Mj_01/templates/example/welcome.html.ep

(bionic)westm@localhost:~/google_drive/mojo$ mojo generate app MyApp
  [mkdir] /home/westm/google_drive/mojo/my_app/script
  [write] /home/westm/google_drive/mojo/my_app/script/my_app
  [chmod] /home/westm/google_drive/mojo/my_app/script/my_app 744
  [mkdir] /home/westm/google_drive/mojo/my_app/lib
  [write] /home/westm/google_drive/mojo/my_app/lib/MyApp.pm
  [exist] /home/westm/google_drive/mojo/my_app
  [write] /home/westm/google_drive/mojo/my_app/my_app.yml
  [mkdir] /home/westm/google_drive/mojo/my_app/lib/MyApp/Controller
  [write] /home/westm/google_drive/mojo/my_app/lib/MyApp/Controller/Example.pm
  [mkdir] /home/westm/google_drive/mojo/my_app/t
  [write] /home/westm/google_drive/mojo/my_app/t/basic.t
  [mkdir] /home/westm/google_drive/mojo/my_app/public
  [write] /home/westm/google_drive/mojo/my_app/public/index.html
  [mkdir] /home/westm/google_drive/mojo/my_app/templates/layouts
  [write] /home/westm/google_drive/mojo/my_app/templates/layouts/default.html.ep
  [mkdir] /home/westm/google_drive/mojo/my_app/templates/example
  [write] /home/westm/google_drive/mojo/my_app/templates/example/welcome.html.ep

(bionic)westm@localhost:~/google_drive/mojo$ mojo generate app WEST_MJ_01
  [mkdir] /home/westm/google_drive/mojo/west__mj_01/script
  [write] /home/westm/google_drive/mojo/west__mj_01/script/west__mj_01
  [chmod] /home/westm/google_drive/mojo/west__mj_01/script/west__mj_01 744
  [mkdir] /home/westm/google_drive/mojo/west__mj_01/lib
  [write] /home/westm/google_drive/mojo/west__mj_01/lib/WEST_MJ_01.pm
  [exist] /home/westm/google_drive/mojo/west__mj_01
  [write] /home/westm/google_drive/mojo/west__mj_01/w_e_s_t__m_j_01.yml
  [mkdir] /home/westm/google_drive/mojo/west__mj_01/lib/WEST_MJ_01/Controller
  [write] /home/westm/google_drive/mojo/west__mj_01/lib/WEST_MJ_01/Controller/Example.pm
  [mkdir] /home/westm/google_drive/mojo/west__mj_01/t
  [write] /home/westm/google_drive/mojo/west__mj_01/t/basic.t
  [mkdir] /home/westm/google_drive/mojo/west__mj_01/public
  [write] /home/westm/google_drive/mojo/west__mj_01/public/index.html
  [mkdir] /home/westm/google_drive/mojo/west__mj_01/templates/layouts
  [write] /home/westm/google_drive/mojo/west__mj_01/templates/layouts/default.html.ep
  [mkdir] /home/westm/google_drive/mojo/west__mj_01/templates/example
  [write] /home/westm/google_drive/mojo/west__mj_01/templates/example/welcome.html.ep

**Comment by Grinnz** [comment, marked as answer]
The convention for Perl package names, and thus application names, is CamelCase. This is then converted to a lowercase script name (and directory name) with underscores using https://metacpan.org/pod/Mojo::Util#decamelize.

**Reply by westmj** [reply]
Excellent explanation, thank you.  snake_case_from_here_on !
